tensorboard/plugins/audio/audio_plugin_test.py

```diff
@@ -18,7 +18,7 @@
 from __future__ import division
 from __future__ import print_function
 
-import collections
+import collections.abc
```

Comment:
Is it necessary to make the change in this line? If you just `import collections`, the instances of `collections.abc.Callable` below will still work right? 
If the purpose is to limit the scope and import only what's used, consider `from collections import abc`.

This comment applies to other files touch in this PR as well.

Reply:
Yes, the change is necessary: `collections.abc` is a submodule, and must
be imported to be available on the `collections` object:

```python
>>> import collections
>>> collections.abc.Callable
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.7/collections/__init__.py", line 55, in __getattr__
    raise AttributeError(f'module {__name__!r} has no attribute {name!r}')
AttributeError: module 'collections' has no attribute 'abc'
```

(Note that `os.path` is a special case: the `os` module itself
explicitly [installs an `os.path` module][1] because it’s dynamically
chosen from `posixpath` or `ntpath` at runtime.)

I considered `from collections import abc`, but decided that it was
confusing given that `abc` is also a standard library module that we use
fairly frequently.

[1]: https://github.com/python/cpython/blob/58a4054760bffbb20aff90290dd0f3554f7bea42/Lib/os.py#L93

Reply:
@wchargin Hmm. Okay, I tested manually in my local ipython and the following worked:

```
In [1]: import collections                                                                                                                                                                             

In [2]: collections.abc                                                                                                                                                                                
Out[2]: <module 'collections.abc' from '/usr/local/google/home/cais/venv_tf_nightly/lib/python3.7/collections/abc.py'>

In [3]: collections.abc.Callable                                                                                                                                                                       
Out[3]: collections.abc.Callable
```

But I guess the tensorboard CI infra may be using a different version of Python.

Reply:
No, you’re using IPython, and the IPython infrastructure has already
imported `collections.abc` at some point, which causes the `abc`
attribute to be assigned onto the `collections` module object. That is,
importing a new submodule has a side effect of mutating the parent
module object.

It’s not a Python version thing. You should be able to reproduce this in
a standard Python shell or Python script.